packages/thirdweb/src/wallets/in-app/core/wallet/enclave-wallet.ts

```diff
@@ -190,13 +196,44 @@ export class EnclaveWallet implements IWebWallet {
           chain: getCachedChain(tx.chainId),
         });
         const signedTx = await _signTransaction(tx);
+
         const transactionHash = await eth_sendRawTransaction(
           rpcRequest,
           signedTx,
         );
-        return {
+
+        // Non-blocking wait for receipt to track transaction
+        waitForReceipt({
```

Comment:
I think I would just track an event like 'transaction_sent' inside sendTransaction, will cover all wallets in one central point.

And we can also track a separate action like 'transaction_confirmed' inside 'waitForReceipt' or 'getTransactionReceipt'.

These two will share the same transaction hash so we can show interesting data for tx sent vs confirmed easily